libvast/src/pattern.cpp

```diff
@@ -116,6 +116,22 @@ bool operator<(const pattern& lhs, const pattern& rhs) {
   return lhs.str_ < rhs.str_;
 }
 
+bool operator==(const pattern& lhs, std::string_view rhs) {
+  return lhs.match(rhs);
+}
+
+bool operator!=(const pattern& lhs, std::string_view rhs) {
+  return !(lhs == rhs);
+}
+
+bool operator==(std::string_view lhs, const pattern& rhs) {
+  return rhs.match(lhs);
```

Comment:
Can you add a TODO for C++20 to remove the symmetric operators?

For now, do you think this should be expressed as `operator==(rhs, lhs)`?

Reply:
I didn't know about this feature in C++20. Great to see that equality is finally symmetric. Until we refactor our code base accordingly, I'd say we leave this as is.